src/convertToObject.js

```diff
@@ -12,7 +12,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  let arr = sourceString.split(';');
+
+  arr = arr.map(string => string.trim());
+  arr = arr.map(string => string.split(':'));
+  arr = arr.map(string => string.map((value) => value.trim()));
```

Comment:
```suggestion
  let arr = sourceString
    .split(';')
    .map(string => string.split(':'))
    .map([key, value] => [key.trim(), value.trim()])
    .filter([key, value] => key && value);
```
Ты можешь запускать методы цепочкой, делать трим в самом конце и фильтровать по длине